lib/internal/streams/readable.js

```diff
@@ -1284,9 +1321,267 @@ ObjectDefineProperties(ReadableState.prototype, {
     set(value) {
       this[kPaused] = !!value;
     }
+  },
+
+  // https://fetch.spec.whatwg.org/#dom-body-bodyused
+  bodyUsed: {
+    get() {
+      return isDisturbed(this);
+    }
+  },
+
+  body: {
+    get() {
+      if (this[kConsume]?.type === kWebStreamType) {
+        return this[kConsume].stream;
+      }
+
+      return consume(this, kWebStreamType);
+    }
+  },
+
+  text: {
+    get() {
+      return consume(this, kTextType);
+    }
+  },
+
+  json: {
+    get() {
+      return consume(this, kJSONType);
+    }
+  },
+
+  blob: {
+    get() {
+      return consume(this, kBlobType);
+    }
+  },
+
+  arrayBuffer: {
+    get() {
+      return consume(this, kArrayBufferType);
+    }
   }
 });
 
+function isLocked(self) {
+  return self[kConsume]?.stream?.locked === true;
+}
+
+// https://streams.spec.whatwg.org/#readablestream-disturbed
+function isDisturbed(self) {
+  return self.destroyed || self.readableDidRead;
+}
+
+// https://fetch.spec.whatwg.org/#body-unusable
+function isUnusable(self) {
+  return isDisturbed(self) || isLocked(self);
+}
+
+function consume(self, type) {
+  if (isUnusable(self)) {
+    throw new TypeError('unusable');
```

Comment:
Who do I ping here? How can I disable this lint error?

Reply:
Why disable it? Can't we reasonably give this error a code?

Reply:
The spec says to throw a `TypeError`.

Reply:
https://fetch.spec.whatwg.org/#concept-body-consume-body